src/app/pages/dashboard/components/dashboard/dashboard.component.ts

```diff
@@ -619,11 +619,21 @@ export class DashboardComponent implements OnInit, AfterViewInit, OnDestroy {
       return;
     }
 
-    const hidden = this.dashState
-      .filter((w) => state.every((s) => !(w.identifier && w.identifier == s.identifier || w.name && w.name == s.name)))
-      .map((widget) => ({ ...widget, rendered: false }));
+    const dashState = this.dashState.map((widget) => {
+      if (state.every((s) => !(
+        widget.identifier
+        && widget.identifier == s.identifier
+        || widget.name && widget.name == s.name
+      ))) {
+        return {
+          ...widget,
+          rendered: false,
+        };
+      }
+      return widget;
+    });
 
-    this.setDashState([...state, ...hidden]);
+    this.setDashState(dashState);
```

Comment:
I noticed that we've lost ability to drag widgets around to reorder them. You can check this feature by clicking "Reorder" button on the dashboard.

Reply:
Sorry, fixed filtering condition